catfs/fs.go

```diff
@@ -955,35 +955,36 @@ func (fs *FS) Stage(path string, r io.Reader) error {
 		fs.cfg.String("compress.default_algo"),
 	)
 
+	// We might not be able to read the header for some reason.
```

Comment:
I would move the definition above the read of the default compression. Otherwise `err` check is visually disconnected from the code producing it.

Reply:
Done.